pkg/loop/internal/pb/ccip/factories.proto

```diff
@@ -21,30 +34,45 @@ message NewExecutionFactoryResponse {
     uint32 execution_factory_service_id = 1;
 }
 
-// ExecutionCustomHandlers is a gRPC service that services the custom
-// resource generation requests for the execution provider
-service ExecutionCustomHandlers {
-    rpc NewOnRampReader(NewOnRampReaderRequest) returns (NewOnRampReaderResponse) {}
+service CommitFactoryGenerator {
+    rpc NewCommitFactory(NewCommitFactoryRequest) returns (NewCommitFactoryResponse);
+}
+
+// NewCommitFactoryRequest is a gRPC adapter to the factory configuration [https://github.com/smartcontractkit/ccip/core/services/ocr2/plugins/ccip/ccipcommit/CommitPluginStaticConfig]
+message NewCommitFactoryRequest {
+    uint32 provider_service_id = 1;
+}
+
+// NewCommitFactoryResponse is a reference to the factory generator service. It is a gRPC proxy mechanism 
+// for the return value of [github.com/smartcontractkit/ccip/core/services/ocr2/plugins/ccip/ccipcommit.NewCommitReportingPluginFactory]
+// because the underlying return value is an interface, which is modeled as a service in gRPC
+message NewCommitFactoryResponse {
+    int32 commit_factory_service_id = 1;
```

Comment:
Is there a deterministic relationship between `provider_service_id` and `commit_factory_service_id`?

Reply:
no